Samples/2.2/GenericHost/src/OrleansClient/Program.cs

```diff
@@ -0,0 +1,88 @@
+﻿using HelloWorld.Interfaces;
+using Orleans;
+using Orleans.Runtime;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Configuration;
+
+namespace OrleansClient
+{
+    /// <summary>
+    /// Orleans test silo client
+    /// </summary>
+    public class Program
+    {
+        const int initializeAttemptsBeforeFailing = 5;
+        private static int attempt = 0;
+
+        static int Main(string[] args)
+        {
+            return RunMainAsync().Result;
+        }
+
+        private static async Task<int> RunMainAsync()
+        {
+            try
+            {
+                using (var client = await StartClientWithRetries())
+                {
+                    await DoClientWork(client);
+                    Console.ReadKey();
+                }
+
+                return 0;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                Console.ReadKey();
+                return 1;
+            }
+        }
+
+        private static async Task<IClusterClient> StartClientWithRetries()
+        {
+            attempt = 0;
+            IClusterClient client;
+            client = new ClientBuilder()
```

Comment:
Why is the client not generated with the generic host interface? The client could live within an mvc app

Reply:
Basically this PR is to make the silo compatible with the new hosting abstractions.

Yes, I do have the client already working but I prefer to put it outside in another PR.

Reply:
BTW, I see the client emulating the HttpClientFactory, since you don't "host" a client.... it wouldn't be uncommmon to have more than 1 client, each client with it's own DI container (it should be almost an implementation detail of the client), and just sharing it's instance (the entry point) with the parent host (ie: ASP.NET Core) container. So basically the client builder wouldn't be replaced with the generic Host (as it should continue to work standalone), but instead there would be some extension methods to have the generic host configure the client factory, similar to how you could use HttpClient without an ASP.NET Core app.

Reply:
Sure, but the client builder should use the same DI container, so that it can share configuration and logging and other services from the host.

Reply:
Configuration and logging are some cross-cutting concerns that could be flowed into the builder from the parent host, without the need for them to share the container. I explained it in length here: #4744

Reply:
Of course, but that is about connection pools for clustering, just as an example